clients/src/test/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumerTest.java

```diff
@@ -1904,6 +1912,44 @@ public void testUnsubscribeWithoutGroupId() {
         verify(applicationEventHandler).add(ArgumentMatchers.isA(UnsubscribeEvent.class));
     }
 
+    @Test
+    public void testSeekToBeginning() {
+        Collection<TopicPartition> topics = Collections.singleton(new TopicPartition("test", 0));
+        time = new MockTime(1);
+        consumer = newConsumer();
+        consumer.seekToBeginning(topics);
+        CompletableApplicationEvent<Void> offsetEvent = addAndGetLastEnqueuedEvent();
+        assertSame(offsetEvent.getClass(), ResetOffsetEvent.class);
+        assertSame(((ResetOffsetEvent) offsetEvent).offsetResetStrategy(), OffsetResetStrategy.EARLIEST);
+    }
+
+    @Test
+    public void testSeekToBeginningWithException() {
+        Collection<TopicPartition> topics = Collections.singleton(new TopicPartition("test", 0));
+        consumer = newConsumer();
+        completeResetOffsetEventExceptionally(new TimeoutException());
+        assertThrows(TimeoutException.class, () -> consumer.seekToBeginning(topics));
+    }
+
+    @Test
+    public void testSeekToEndWithException() {
+        Collection<TopicPartition> topics = Collections.singleton(new TopicPartition("test", 0));
+        consumer = newConsumer();
+        completeResetOffsetEventExceptionally(new TimeoutException());
+        assertThrows(TimeoutException.class, () -> consumer.seekToEnd(topics));
+    }
+
+    @Test
+    public void testSeekToEnd() {
+        Collection<TopicPartition> topics = Collections.singleton(new TopicPartition("test", 0));
+        time = new MockTime(1);
```

Comment:
harmless but not needed I guess? We usually need an instance of `MockTime` when we need to advance time for the test, but here probably just `mock(Time.class)` would be enough.

Reply:
Thanks for review, all comments are addressed.